controllers/topics/create.js

```diff
@@ -0,0 +1,42 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    await Topics.create({
+      name,
+      icon_path: '',
+      is_custom_topic: true,
+      created_at: new Date(),
+      categories: ''
+    });
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success'
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic of the code seems to have a flaw. If the topic already exists, it sends a response back but does not stop the execution of the function. This means that even if the topic already exists, the function will still try to create a new one with the same name, which could lead to an error or unexpected behavior. To fix this, you should add a `return` statement after sending the response when the topic already exists.

```diff
    if (topics) {
      res.status(400).json({
        code: 400,
        status: 'failed',
        message: 'Topic already exists'
      });
+       return;
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->